orderbook/src/routes/mod.rs

```diff
@@ -0,0 +1,161 @@
+mod session;
+pub mod websocket;
+
+use crate::models::Order;
+use crate::routes::websocket::PriceFeed;
+use crate::routes::websocket::UpdatedOrder;
+use actix::Addr;
+use actix_web::delete;
+use actix_web::get;
+use actix_web::http::header::ContentType;
+use actix_web::post;
+use actix_web::put;
+use actix_web::web;
+use actix_web::Error;
+use actix_web::HttpRequest;
+use actix_web::HttpResponse;
+use actix_web_actors::ws;
+use diesel::r2d2;
+use diesel::r2d2::ConnectionManager;
+use diesel::PgConnection;
+use serde::Deserialize;
+use serde::Serialize;
+use std::time::Instant;
+
+#[derive(Deserialize, Serialize)]
+pub struct NewOrder {
+    pub price: i32,
+    pub maker_id: String,
+    pub taken: bool,
+}
+
+impl From<NewOrder> for crate::models::NewOrder {
+    fn from(value: NewOrder) -> Self {
+        crate::models::NewOrder {
+            price: value.price,
+            maker_id: value.maker_id,
+            taken: value.taken,
+        }
+    }
+}
+
+#[derive(Deserialize, Serialize)]
+pub struct UpdateOrder {
+    pub taken: bool,
+}
+
+type DbPool = r2d2::Pool<ConnectionManager<PgConnection>>;
+
+/// Finds order by UID.
+#[get("/orders/{order_id}")]
+pub async fn get_order(
+    pool: web::Data<DbPool>,
+    order_id: web::Path<i32>,
+) -> Result<HttpResponse, Error> {
+    let order_uid = order_id.into_inner();
+
+    // use web::block to offload blocking Diesel code without blocking server thread
+    let order = web::block(move || {
+        let mut conn = pool.get().unwrap();
+        Order::get_with_id(&mut conn, order_uid)
+    })
+    .await?
+    .map_err(actix_web::error::ErrorInternalServerError)?;
+
+    if let Some(order) = order {
+        Ok(HttpResponse::Ok().json(order))
+    } else {
+        let res = HttpResponse::NotFound().body(format!("No order found with id: {order_uid}"));
+        Ok(res)
+    }
+}
+
+/// Get all orders
+#[get("/orders")]
+pub async fn get_orders(pool: web::Data<DbPool>) -> Result<HttpResponse, Error> {
+    // use web::block to offload blocking Diesel code without blocking server thread
```

Comment:
This is something we need to be very careful of in general; in the past I've gone as far as only allowing access to the DB through a `database` model which wraps every DB call in `async` first. Otherwise maybe we can lint or something. It is just very easy to forget one DB call on the reactor thread and then get weird performance issues elsewhere (as seen in ItchySats)